### ISSUE TITLE
Specific model version(or names) for DINO and CLIP

### ISSUE THREAD

**Issue by WonderStarsNight** [LuizScarlet/StableCodec#6]
StableCodec is a very effective and surprisingly impressive piece of work. Thank you for open-sourcing it. I am trying to reproduce it—could you please provide the specific model version( or names) for DINO and CLIP?

**Comment by LuizScarlet** [comment]
Hi @WonderStarsNight , we use `openai/clip-vit-base-patch32` from Hugging Face and `dinov2_vitb14_reg4_pretrain.pth` from https://github.com/facebookresearch/dinov2.

**Comment by LuizScarlet** [comment]
Sorry I missed it previously. For your questions:
1. Yes, we directly resize from 512 to 224. We find this simple implementation works quite well under our tasks compared to other workaround, though it might be crude.
2. We adopt BCE as it is more stable in our training.

**Comment by WonderStarsNight** [comment]
I really appreciate it!! Thank you for allowing me to ask these questions! And, could you please describe the preprocessing applied to your dataset? Specifically, how were the images processed—how were they resized and cropped?

**Comment by WonderStarsNight** [comment]
Thank you very much for your help. I have reproduced stage 1 and conducted real-time testing on the Kodak dataset. The test results for PSNR, LPIPS, and other metrics are aligned with the stage 1 logs you provided. However, when we use the 100k step checkpoint from stage 1 to train stage 2, in the FT2 experiment, we found that the PSNR and LPIPS do not align with the stage 2 logs. Our LPIPS is 0.20, which is higher compared to your reported LPIPS of 0.17.

Regarding the discriminator model design, we used the official DinoV2 model and the corresponding weights you provided as the backbone of the discriminator. For the head, we adopted the decoder module from Vision-Aided GAN, using an MLPD with input channels = 768 and output channels = 256 for training. Please refer to the following figure  or https://github.com/nupurkmr9/vision-aided-gan/blob/95fc55beefad3e868783beab421108c5baf583aa/vision_aided_loss/cv_discriminator.py#L74 for details. For the discriminator's forward pass, we directly use the feature outputs from DinoV2's forward method and input them into the head. We use these features to compute the BCE loss.

As for the GAN training strategy, we employ a two-step training approach:

Step 1: Optimize the main network using optimizer1 with adv_loss = real_loss combined with reconstruction loss R and discriminator loss D.
Step 2: Optimize only the discriminator using optimizer2 (which has the same settings as optimizer1), with adv_loss = (fake_loss + real_loss) / 2.
Could you please share how your head is designed and how you select the intermediate features as inputs to the head? Additionally, is your training strategy the same?

<img width="819" height="370" alt="Image" src="https://github.com/user-attachments/assets/9c000600-1204-4b44-adc0-ac34ca02d288" />
